README.md

```diff
@@ -1,10 +1,104 @@
-Authors: Aaron Alvarez, Akshat Chaturvedi, Ethan Reed
 # Project 2: Simple Bulletin Board Using Socket Programming
-### Team Members
+### Team Members 
+
 **Akshat Chaturvedi**
 **Aaron Alvarez**
 **Ethan Reed**
 
 ## Introduction
 
+This project implements a client-server bulletin board system using unicast sockets. The application consists of two parts:
+
+1. A public message board where users can join, post messages, and interact with others.
+2. Private message boards allowing users to join multiple groups and exchange group-specific messages.
+
+## Features
+### Part 1: Public Message Board
+
+**Users can:**
+
+1. Connect to the server using a unique username.
+2. Post public messages visible to all users.
+3. Retrieve the last two messages posted before joining.
+4. View a list of connected users.
+5. Retrieve specific messages by ID.
+6. Leave the group.
+
+### Part 2: Private Message Boards
+
+**Users can:**
+
+1. View and join available groups.
+2. Participate in multiple groups simultaneously.
+3. Post and retrieve group-specific messages.
+4. View group members.
+5. Leave individual groups.
+
+## Technologies Used
+
+**Programming Languages :** Python
+
+**Sockets :** TCP sockets for communication
+
+**Protocol :** Custom-designed
+
+## Commands and Usage
+
+### General Commands
+
+%connect 127.0.0.1 65432 : Connect to the server.
```

Comment:
for the %connect command, we should also specify that the ip address should be that of the server, and the port should be the port to connect to. since the ip address and port can change depending on the servers configs